sormas-backend/src/main/java/de/symeda/sormas/backend/externalmessage/ExternalMessageFacadeEjb.java

```diff
@@ -201,6 +212,54 @@ public ExternalMessageDto save(@Valid ExternalMessageDto dto, boolean checkChang
 		validate(dto);
 
 		externalMessage = fillOrBuildEntity(dto, externalMessage, checkChangeDate);
+
+		// If it is a LabMessage and it has not set a DiseaseVariant, an attempt is made to determine this from the attached TestReports.
+		if (ExternalMessageType.LAB_MESSAGE.equals(externalMessage.getType())
```

Comment:
I don't see what happens here it got too complicated

You just have to get all variants and all variant details into sets and if those have only a single element than you set the disease variant and details on the lab message